### DISCUSSION TITLE
PANIC/EXCEPTION/STACKTRACE not available in Log Exporter Records

### DISCUSSION THREAD

**Discussion by js10x** [open-telemetry/opentelemetry-go#5889] [Q&A]
PANIC/EXCEPTION/STACKTRACE not available in logged record type:
https://pkg.go.dev/go.opentelemetry.io/otel/sdk/log@v0.7.0#Record

Is there any way to access any kind of exception/panic or stack trace from the log records? I can't find it in the above link for the type. FYI: This is the record type used when using the log exporter.

**Comment by dmathieu** [comment]
Log records have a severity level, which include error and fata. Is this what you're looking for?
https://pkg.go.dev/go.opentelemetry.io/otel/log#Severity

Log bridges transform the log level into this severity.
See the doc for otelslog for example: https://pkg.go.dev/go.opentelemetry.io/contrib/bridges/otelslog#hdr-Record_Conversion

**Reply by js10x** [reply]
So, you're saying that if a panic occurred, converting the severity text to a string will show the stack trace for the panic?

**Reply by dmathieu** [reply]
No. The SDK will not catch `panics`, and that's not its purpose.
The logs SDK provides bridges for logging libraries to emit data that is correlated with OpenTelemetry, and can then be linked to traces/metrics, and exporter in OTLP format.
We don't provide any mean to catch panics and recover their stacktraces though. That is something the logging library would have to do.

If a logging library emits a log entry with a severity level equivalent of panic, that severity level will be propagated through the otel log bridge.

**Reply by pellared** [reply]
Here are the [Semantic Conventions for Exceptions in Logs](https://opentelemetry.io/docs/specs/semconv/exceptions/exceptions-logs/) that you should follow when logging panics.

**Reply by pellared** [reply]
Related OTEP:
- https://github.com/open-telemetry/opentelemetry-specification/pull/4333
